src/player/index.ts

```diff
@@ -232,11 +302,36 @@ function getNextTour(): CodeTour | undefined {
   }
 }
 
-function getHeaderText(step: CodeTourStep): string {
+async function getFileContext(
+  line: number,
+  file: string,
+  uri: Uri
+): Promise<string> {
+  let label = `This step is on line ${line} in file ${file}`;
+  const document = await workspace.openTextDocument(uri);
+  if (document.languageId === "typescript") {
+    let tokens = await tokenize(document.getText().split("\n"));
+    let index = line - 1;
+    while (index < tokens.length && index >= 0) {
```

Comment:
Whoops, that's a bad bug. I had `--index;` in there in an earlier version, and then it vanished... But the loop is to make sure if you have a CodeTour step in the middle of a function, it can find the nearest parent scope.